### PR TITLE
Fix: lists design system [NP-1074] 

### PR DESCRIPTION
**Pull request by davidsauntson**
- Removes list utility classes from the design-system, with the exception of `.cads-list-no-bullet` which is widely used in the public-website repo
- Remove uses of `.cads-list--no-bullet` from design-system
- Align list markers with grid in prose content
- Fix issue with paragraphs in list items from migrated content

### CODE REVIEW

File: scss/4-elements/_lists.scss [hunk 1]
```diff
@@ -1,5 +1,4 @@
-li,
```
Comment (davidrapson) [inline scss/4-elements/_lists.scss:1]: ❤️

File: scss/4-elements/_lists.scss [hunk 2]
```diff
@@ -10,42 +9,80 @@ li,
...
+  margin: 0 0 $cads-spacing-4 0;
+  list-style-type: none;
+
+  li {
+    display: inline-block;
+    margin-right: 1rem;
+  }
+
+  // Reverse margins if this is right aligned
+  .cads-align-right & {
```
Comment (davidrapson) [inline scss/4-elements/_lists.scss:42]: Is this only used in the header? If so maybe we should inline these styles there?

File: scss/4-elements/_lists.scss [hunk 3]
```diff
@@ -10,42 +9,80 @@ li,
...
+  li {
+    @extend %cads-list-item;
+  }
+}
+
+.cads-list-no-bullet {
+  @extend %cads-list-no-bullet;
+}
+
+.cads-prose {
```
Comment (davidrapson) [inline scss/4-elements/_lists.scss:64]: Style preference: Does it make sense to have a single file for the prose styles, or happy defining this inline like this?

File: scss/6-components/advisernet/_related-content.scss [hunk 4]
```diff
@@ -66,3 +66,11 @@
     }
   }
 }
+
+// :-( this whole file is a bit much
```
Comment (davidrapson) [inline scss/6-components/advisernet/_related-content.scss:70]: Yeah, fine for this to be a bit gnarly for now as we're going to pull this out anyway

File: styleguide/sample-pages/advice-collection/_advice_collection_adviser.html.haml [hunk 5]
```diff
@@ -27,6 +27,26 @@
                 %p The important callout should be used for any important snippet of text that has serious and/or legal implications if the client does not follow the advice.
             %p
               Every page should have some lorem ipsum. So here it is.
+            %ul
```
Comment (davidrapson) [inline styleguide/sample-pages/advice-collection/_advice_collection_adviser.html.haml:30]: Maybe we should create a "kitchen sink" sample page with this and a few more prose specific examples?

**Review by davidrapson — APPROVED** [review]
👍 Nice! A few comments but nothing blocking (although unsure if anything changes based on your mixin suggestion?)

**Comment by davidsauntson** [comment]
Thanks! - I think I will do the mixin stuff as a second PR and work on the advisernet template stuff first.  That way there won't be another breaking change later on 👍

File: scss/4-elements/_lists.scss (hunk 3 above)
Comment (davidsauntson) [reply scss/4-elements/_lists.scss:64]: yup good idea, I've made a new objects file `_prose` in section 5 👍

File: styleguide/sample-pages/advice-collection/_advice_collection_adviser.html.haml (hunk 5 above)
Comment (davidsauntson) [reply styleguide/sample-pages/advice-collection/_advice_collection_adviser.html.haml:30]: I think the public advice collection one covers most things, I've added lists to that one too.  We can add more tho!

File: scss/5-objects/_prose.scss [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    li p {
+      margin-bottom: 0;
+    }
+  }
+}
+
+.cads-prose {
+  @extend %cads-prose;
+
+  .cads-list--no-bullet {
```
Comment (davidrapson) [inline scss/5-objects/_prose.scss:46]: Does this need to be `cads-list-no-bullet` too?

Comment (davidsauntson) [reply scss/5-objects/_prose.scss:46]: 🤦

**Review by soniaturcotte — APPROVED** [review]
LGTM

**Review by luke-hill — APPROVED** [review]

**Pull request merged**